docs/codecs.rst

```diff
@@ -1,11 +1,207 @@
+<https://github.com/zarr-developers/zarr-specs/labels/codec>`_.
+
+This document is maintained by the `Zarr core development team
+<https://github.com/orgs/zarr-developers/teams/core-devs>`_.
+
+
+Document conventions
+====================
+
+This document lists a collection of codecs. For each codec, the
+following information is provided:
+
+* A URI which can be used to uniquely identify the codec in Zarr array
+  metadata.
+* Any configuration parameters which can be set in Zarr array
+  metadata.
+* A definition of encoding/decoding algorithm and the encoded format,
+  or a citation to an existing specification where this is defined.
+* Any additional headers added to the encoded data.
+
+Conformance requirements are expressed with a combination of
+descriptive assertions and [RFC2119]_ terminology. The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in the normative
+parts of this document are to be interpreted as described in
+[RFC2119]_. However, for readability, these words do not appear in all
+uppercase letters in this specification.
+
+All of the text of this specification is normative except sections
+explicitly marked as non-normative, examples, and notes. Examples in
+this specification are introduced with the words "for example".
+
+
 Codecs
 ======
 
-Under construction.
+Gzip
+----
+
+Codec URI:
+    https://purl.org/zarr/spec/codecs/gzip
+
+    
+Configuration parameters
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+level:
+    An integer from 0 to 9 which controls the speed and level of
+    compression. A level of 1 is the fastest compression method and
+    produces the least compressions, while 9 is slowest and produces
+    the most compression. Compression is turned off completely when
+    level is 0.
+
+For example, the array metadata below specifies that the compressor is
+the Gzip codec configured with a compression level of 1::
+
+    {
+        "compressor": {
+            "codec": "https://purl.org/zarr/spec/codecs/gzip",
+            "configuration": {                                                                                
+                "level": 1                                                                                    
+            }
+        },
+    }
+
+    
+Format and algorithm
+~~~~~~~~~~~~~~~~~~~~
+
+Encoding and decoding is performed using the algorithm defined in
+[RFC1951]_.
+
+Encoded data should conform to the Gzip file format [RFC1952]_.
+
+
+Blosc
+-----
+
+Codec URI:
+    https://purl.org/zarr/spec/codecs/blosc
+
+    
+Configuration parameters
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+cname:
+    A string identifying the internal compression algorithm to be
+    used. At the time of writing, the following values are supported
+    by the c-blosc library: "lz4", "lz4hc", "blosclz", "zstd",
+    "snappy", "zlib".
+    
+clevel:
+    An integer between 1 and 9 indicating the compression level.
```

Comment:
According to the [c-blosc source file](https://github.com/Blosc/c-blosc/blob/master/blosc/blosc.h#L147-L148), the compression level can also be 0, in which case there is no compression.
I think we should copy/paste the description of the `level` parameter from GZip, as the relation between compression level and speed of compression probably applies to every underlying compression algorithm.
```suggestion
    An integer from 0 to 9 which controls the speed and level of
    compression. A level of 1 is the fastest compression method and
    produces the least compressions, while 9 is slowest and produces
    the most compression. Compression is turned off completely when
    level is 0.
```